### PR TITLE
Josselin/file uploads

### PR DESCRIPTION
**Pull request by JosselinMorau**
add default image dialog and view for messenger adapter.
Any files attachment will trigger the message :'File uploads are not supported'
task : https://app.asana.com/0/370953015396593/737691243818533/f

### CODE REVIEW

File: src/views/image-view.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { View, BotTextMessage } = require('botfuel-dialog');
+
+class ImageView extends View {
```
Comment (Benjamin15) [inline src/views/image-view.js]: Maybe FileView will be more explicit. You trigger this one when you receive a file, not necessary a image

File: src/dialogs/image-dialog.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const { BaseDialog } = require('botfuel-dialog');
+
+class ImageDialog extends BaseDialog {}
```
Comment (Benjamin15) [inline src/dialogs/image-dialog.js]: Maybe FileDialog will be more explicit. You trigger this one when you receive a file, not necessary a image

**Comment by JosselinMorau** [comment]
I'll check that.

File: src/dialogs/image-dialog.js (hunk 2 above)
Comment (KevinDepuydt) [reply src/dialogs/image-dialog.js]: @Benjamin15 @JosselinMorau maybe there is two dialogs to implement, not only ImageDialog but also an AttachmentDialog or as you said @Benjamin15 a FileDialog.
We need to validate if we want to have different dialogs for images and others attachments types a user can send to the bot or use only one dialog for each attachment types.

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 node_modules
 .DS_Store
 yarn-error.log
-package-lock.json
\ No newline at end of file
+package-lock.json
+jest_0
```
Comment (windkomo) [inline .gitignore]: what is jest_0?

File: package.json [hunk 4]
```diff
@@ -2,6 +2,9 @@
   "name": "botfuel-module-adapter-messenger",
   "version": "1.2.9",
   "main": "index.js",
+  "jest": {
+    "testURL": "http://localhost"
```
Comment (windkomo) [inline package.json]: what is this?

File: src/views/file-view.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const { View, BotTextMessage } = require('botfuel-dialog');
+
+class FileView extends View {
+  render(userMessage, {}) {
+    const botResponses = [new BotTextMessage('File uploads are not supported.')];
```
Comment (windkomo) [inline src/views/file-view.js]: you can just return `[new BotTextMessage('File uploads are not supported.')];`

File: src/views/image-view.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+  render(userMessage, {}) {
+    const botResponses = [new BotTextMessage('Image uploads are not supported.')];
```
Comment (windkomo) [inline src/views/image-view.js]: you can just return `[new BotTextMessage('Image uploads are not supported.')];`

File: test/messenger-adapter.test.js [hunk 7]
```diff
@@ -219,3 +221,33 @@ describe('MessengerAdapter request handling', () => {
     expect(requestStub.firstCall.args).toEqual([expectedBotResponse]);
   });
 });
+
+describe('Test bot answers to uploads', () => {
+  test('Bot\'s response by default to image', async () => {
```
Comment (windkomo) [inline test/messenger-adapter.test.js]: Use `’` (apostrophe) instead of escaping quotes

File: test/messenger-adapter.test.js [hunk 8]
```diff
@@ -219,3 +221,33 @@ describe('MessengerAdapter request handling', () => {
...
+    await bot.play([
+      new UserImageMessage('https://image1.jpg'),
+    ]);
+    expect(bot.adapter.log).toEqual([
+      new UserImageMessage('https://image1.jpg'),
+      new BotTextMessage('Image uploads are not supported.'),
+    ].map(o => o.toJson(userId)));
+  });
+
+  test('Bot\'s response by default to file', async () => {
```
Comment (windkomo) [inline test/messenger-adapter.test.js]: same

File: test/messenger-adapter.test.js [hunk 9]
```diff
@@ -219,3 +221,33 @@ describe('MessengerAdapter request handling', () => {
     expect(requestStub.firstCall.args).toEqual([expectedBotResponse]);
   });
 });
+
+describe('Test bot answers to uploads', () => {
+  test('Bot\'s response by default to image', async () => {
+    const bot = new Bot({adapter:{
```
Comment (windkomo) [inline test/messenger-adapter.test.js]: formatting:
```javascript
const bot = new Bot({
  adapter: {
      name: 'test'
  }
});
```

File: test/messenger-adapter.test.js [hunk 10]
```diff
@@ -219,3 +221,33 @@ describe('MessengerAdapter request handling', () => {
...
+      new UserImageMessage('https://image1.jpg'),
+    ]);
+    expect(bot.adapter.log).toEqual([
+      new UserImageMessage('https://image1.jpg'),
+      new BotTextMessage('Image uploads are not supported.'),
+    ].map(o => o.toJson(userId)));
+  });
+
+  test('Bot\'s response by default to file', async () => {
+    const bot = new Bot({adapter:{
```
Comment (windkomo) [inline test/messenger-adapter.test.js]: same

File: test/messenger-adapter.test.js [hunk 11]
```diff
@@ -219,3 +221,33 @@ describe('MessengerAdapter request handling', () => {
...
+      name: 'test',
+    }});
+    const { userId } = bot.adapter;
+    await bot.play([
+      new UserFileMessage('https://fichier1.txt'),
```
Comment (windkomo) [inline test/messenger-adapter.test.js]: No french plz

File: test/messenger-adapter.test.js [hunk 12]
```diff
@@ -219,3 +221,33 @@ describe('MessengerAdapter request handling', () => {
...
+    ]);
+    expect(bot.adapter.log).toEqual([
+      new UserFileMessage('https://fichier1.txt'),
```
Comment (windkomo) [inline test/messenger-adapter.test.js]: same

File: package.json (hunk 4 above)
Comment (windkomo) [reply package.json]: Updating to the latest jest version fixes it

**Comment by windkomo** [comment]
@JosselinMorau looks 👍 to me, just need to update the jest version to fix the test URL thing

**Pull request merged**